analyzers/src/SonarAnalyzer.CSharp/Rules/RedundantDeclaration.cs

```diff
@@ -430,9 +424,7 @@ private static void ReportIssueOnRedundantObjectCreation(SonarSyntaxNodeReportin
             var location = node is ObjectCreationExpressionSyntax objectCreation
                 ? objectCreation.CreateLocation(objectCreation.Type)
                 : node.GetLocation();
-            context.ReportIssue(Diagnostic.Create(Rule, location,
-                ImmutableDictionary<string, string>.Empty.Add(DiagnosticTypeKey, redundancyType.ToString()),
-                message));
+            context.ReportIssue(Rule, location, ImmutableDictionary<string, string>.Empty.Add(DiagnosticTypeKey, redundancyType.ToString()), message);
```

Comment:
The best I could come up with is to make this a bit easier is:
```cs
// Another overload
public void ReportIssue(DiagnosticDescriptor rule, Location location, KeyValuePair<string, string> property, params string[] messageArgs) => ..

// Named argument is needed to disambiguate
context.ReportIssue(Rule, location, property: new(DiagnosticTypeKey, redundancyType.ToString()), message);
```

or using `Dictionary<string, string> properties` and dictionary initializers. Both options are not very pleasent.

Reply:
Let's wait for C# 13: #9354